shell/platform/embedder/embedder_render_target.cc

```diff
@@ -4,24 +4,38 @@
 
 #include "flutter/shell/platform/embedder/embedder_render_target.h"
 
+#include <optional>
 #include <utility>
 
 #include "flutter/fml/logging.h"
 
+#include "impeller/aiks/aiks_context.h"
+#include "impeller/renderer/render_target.h"
 #include "third_party/skia/include/core/SkSurface.h"
 
 namespace flutter {
 
-EmbedderRenderTarget::EmbedderRenderTarget(FlutterBackingStore backing_store,
-                                           sk_sp<SkSurface> render_surface,
-                                           fml::closure on_release)
+EmbedderRenderTarget::EmbedderRenderTarget(
+    FlutterBackingStore backing_store,
+    sk_sp<SkSurface> render_surface,
+    std::shared_ptr<impeller::AiksContext> aiks_context,
+    std::optional<impeller::RenderTarget> impeller_target,
+    fml::closure on_release)
     : backing_store_(backing_store),
       render_surface_(std::move(render_surface)),
+      aiks_context_(std::move(aiks_context)),
+      impeller_target_(std::move(impeller_target)),
       on_release_(std::move(on_release)) {
   // TODO(38468): The optimization to elide backing store updates between frames
   // has not been implemented yet.
   backing_store_.did_update = true;
-  FML_DCHECK(render_surface_);
+
+  if (impeller_target_) {
+    FML_DCHECK(aiks_context_);
+    FML_DCHECK(!render_surface_);
+  } else {
+    FML_CHECK(render_surface_);
```

Comment:
`DCHECK` turned to `CHECK`. Perhaps keep them the same?

Reply:
Fixed.